scripts/qa_crowbarsetup.sh

```diff
@@ -462,13 +464,53 @@ function cluster_node_assignment()
     wait_for 40 15 "! cat *.chef-client.ret | grep -qv '^0$'" "all chef-clients to succeed" \
         "cat *.chef-client.ret ; complain 73 'Manually triggered chef-client run failed on at least one node.'"
 
-    # assign nodes to clusters
-    clusternodesnetwork=`printf  "%s\n" $L | head -n$nodenumbernetworkcluster`
-    clusternodesservices=`printf "%s\n" $L | tail -n$nodenumberservicescluster`
+    ### Examples for clusterconfig:
+    # clusterconfig="data+services+network=2"
+    # clusterconfig="services+data=2:network=3:::"
+    # clusterconfig="services=3:data=2:network=2:"
+
+    for cluster in ${clusterconfig//:/ } ; do
+        [ -z "$cluster" ] && continue
+        local group=${cluster%=*}
+        local number=${cluster#*=}
+
+        local clustername=${group%%+*}
+        local nodes=
+
+        # clusternodesdata can only be the drbd nodes
+        if [[ $group =~ data ]] ; then
+            nodes="$clusternodesdrbd"
+        fi
+
+        # fetch nodes if not yet defined
+        if [ -z "$nodes" ] ; then
+            nodes=`printf  "%s\n" $L | head -n$number`
+        fi
 
-    for n in $clusternodesnetwork $clusternodesservices ; do
-        L=`printf "%s\n" $L | grep -iv $n`
+        # remove the nodes from the list of available nodes
+        for onenode in $nodes ; do
+            L=`printf "%s\n" $L | grep -iv $onenode`
```

Comment:
> and could we also put printf "%s\n" $L into a separate function
We definitely should not include the $L in this function, to make it reusable for other variables.
The format string is also variable, as it changes the output. So a general function therefore boils down to replace `printf $1 $2` with `printhelper $1 $2`. This does not help.

Creating a specialized function to just replace 
`printf "%s\n" $L`
with
`line_by_line $L`
IMHO does not help us with refactoring either. The goal is to get rid of exactly such bash helpers whith using a real scripting language. Creating such functions now will increase the refactoring work when switching to another language.

> we use it a few times,
2 times :)